src/Elastic.Apm/DistributedTracing/TraceParent.cs

```diff
@@ -0,0 +1,199 @@
+				if (versionArray[0] == 255)
+					return false;
+
+				if (versionArray[0] > 0)
+					// expected version is 00
+					// for higher versions - best attempt parsing of trace id, span id, etc.
+					bestAttempt = true;
+			}
+			catch
+			{
+				return false;
+			}
+
+			if (traceParentValue.Length < VersionAndTraceIdLength || traceParentValue[VersionAndTraceIdLength - 1] != '-') return false;
+
+			try
+			{
+				//parse value to byte array and back to string:
+				//TODO: probably we should just validate the chars
+				var traceIdVal = ByteArrayToString(StringToByteArray(traceParentValue, VersionPrefixIdLength, TraceIdLength));
+
+				if (traceIdVal == "00000000000000000000000000000000")
+					return false;
+
+				traceId = traceIdVal;
+			}
+			catch (ArgumentOutOfRangeException)
+			{
+				return false;
+			}
+
+			if (traceParentValue.Length < VersionAndTraceIdAndSpanIdLength
+				|| traceParentValue[VersionAndTraceIdAndSpanIdLength - 1] != '-') return false;
+
+			try
+			{
+				//parse value to byte array and back to string:
+				//TODO: probably we should just validate the chars
+				var prentIdVal = ByteArrayToString(StringToByteArray(traceParentValue, VersionAndTraceIdLength, SpanIdLength));
+
+				if (prentIdVal == "0000000000000000")
+					return false;
+
+				prentId = prentIdVal;
+			}
+			catch (ArgumentOutOfRangeException)
+			{
+				return false;
+			}
+
+			if (traceParentValue.Length < VersionAndTraceIdAndSpanIdLength + OptionsLength) return false;
+
+			try
+			{
+				traceOptions = StringToByteArray(traceParentValue, VersionAndTraceIdAndSpanIdLength, OptionsLength);
+			}
+			catch (ArgumentOutOfRangeException)
+			{
+				return false;
+			}
+
+			if (!bestAttempt && traceParentValue.Length != VersionAndTraceIdAndSpanIdLength + OptionsLength) return false;
+
+			// ReSharper disable once InvertIf - imo that would make it hard to read.
+			if (bestAttempt)
+			{
+				if (traceParentValue.Length > VersionAndTraceIdAndSpanIdLength + OptionsLength &&
+					traceParentValue[VersionAndTraceIdAndSpanIdLength + OptionsLength] != '-')
+					return false;
+			}
+
+			return true;
+		}
+
+		private static string ByteArrayToString(IEnumerable<byte> bytes)
+		{
+			var sb = new StringBuilder();
+			foreach (var t in bytes) sb.Append(ByteToHexCharArray(t));
+
+			return sb.ToString();
+
+			char[] ByteToHexCharArray(byte b)
+			{
+				var result = new char[2];
+
+				result[0] = (char)ByteToHexLookupTable[b];
+				result[1] = (char)(ByteToHexLookupTable[b] >> 16);
+
+				return result;
+			}
+		}
+
+		private static readonly uint[] ByteToHexLookupTable = CreateLookupTable();
+
+		// https://stackoverflow.com/a/24343727
+		private static uint[] CreateLookupTable()
+		{
+			var table = new uint[256];
+			for (var i = 0; i < 256; i++)
+			{
+				var s = i.ToString("x2");
+				table[i] = s[0];
+				table[i] += (uint)s[1] << 16;
+			}
+
+			return table;
+		}
+
+		public static string GetTraceParentVal(string traceId, string spanId)
```

Comment:
I think `BuildTraceParentVal` better reflects what the method does :)

Reply:
Agreed. Changed it.